mlflow/server/js/src/common/utils/Utils.tsx

```diff
@@ -276,12 +276,15 @@ class Utils {
     const bitbucketMatch = sourceName.match(Utils.getBitbucketRegex());
     const gitMatch = sourceName.match(Utils.getGitRegex());
     let url = null;
-    if (gitHubMatch || gitLabMatch) {
-      const baseUrl = gitHubMatch ? 'https://github.com/' : 'https://gitlab.com/';
-      const match = gitHubMatch || gitLabMatch;
-      url = baseUrl + match[1] + '/' + match[2].replace(/.git/, '');
-      if (match[3]) {
-        url = url + '/tree/master/' + match[3];
+    if (gitHubMatch) {
+      url = 'https://github.com/' + gitHubMatch[1] + '/' + gitHubMatch[2].replace(/.git/, '');
+      if (gitHubMatch[3]) {
+        url = url + '/tree/master/' + gitHubMatch[3];
+      }
+    } else if (gitLabMatch) {
+      url = 'https://gitlab.com/' + gitLabMatch[1] + '/' + gitLabMatch[2].replace(/.git/, '');
+      if (gitLabMatch[3]) {
+        url = url + '/-/tree/master' + gitLabMatch[3];
```

Comment:
```suggestion
        url = url + '/-/tree/master/' + gitLabMatch[3];
```

Reply:
The regex for gitLabMatch[3] doesn't remove the / at the start, so this could become something like `/-/tree/master//file.py` for example. Is that the desired behavior?